starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java

```diff
@@ -56,7 +56,23 @@ public void createTest() {
     Assert.assertTrue(inventory1.getName() == NAME);
     Assert.assertTrue(inventory1.getProductType() == PRODUCT_TYPE);
   }
-
+  /**
+   * Test delete method
+   */
+  @Test
+  public void deleteTest() {
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setId(ID);
+    inventory.setProductType(PRODUCT_TYPE);
+    inventory = this.mongoTemplate.save(inventory);
+    String id = inventory.getId();
+    Inventory deletedInventory = inventoryDAO.delete(id).get();
```

Comment:
The point of returning an optional is that you would use that container to check for the presence of the object inside. Take a look at the Optional javadoc to see if you can find a method that would tell you if the optional has something in it.

Reply:
Also, calling get an an optional that has nothing in it will cause an exception to be thrown, which is not what we want here.